src/html/boldOrItalic.js

```diff
@@ -4,7 +4,127 @@ var strings = require('../strings');
 var wrapping = require('./wrapping');
 
 function boldOrItalic (chunks, type) {
-  wrapping(type === 'bold' ? 'strong' : 'em', strings.placeholders[type], chunks);
+  wrapping_bold_italic(type === 'bold' ? 'strong' : 'em', strings.placeholders[type], chunks);
+}
+
+function wrapping_bold_italic(tag, placeholders, chunks){
+  var open = '<' + tag;                                            
+  var close = '</' + tag.replace(/</g, '</');                    
+  var rleading = new RegExp(open + '( [^>]*)?>$', 'i');            
+  var rtrailing = new RegExp('^' + close + '>', 'i');              
+  var ropen = new RegExp(open + '( [^>]*)?>', 'ig');               
+  var rclose = new RegExp(close + '( [^>]*)?>', 'ig');                  
+  
+  chunks.trim();
+
+  var trail = rtrailing.exec(chunks.after);
+  var lead = rleading.exec(chunks.before);
+  if (lead && trail) {
+    chunks.before = chunks.before.replace(rleading, '');
+    chunks.after = chunks.after.replace(rtrailing, '');
+  }
+  else if(!chunks.selection) {
+  	if(!trail && !lead && surrounded(chunks,tag)){
+  		chunks.selection = placeholders;
+        chunks.after = close + '>' + chunks.after;
+        chunks.before += open + '>';
+  	}
+  	else if(!trail && !lead surrounded(chunks, tag)) {
```

Comment:
Getting error here./
I think it should be `  	else if(!trail && !lead && surrounded(chunks, tag)) {  `